### PR TITLE
CMake Adol-C detection

### PR DESCRIPTION
**Pull request by jppelteret**
Also includes some stand-alone tests to ensure that Adol-C is indeed detected and is functional.

### CODE REVIEW

File: cmake/configure/configure_adolc.cmake [hunk 1]
```diff
@@ -0,0 +1,61 @@
...
+      )
+
+    IF(EXISTS ${ADOLC_SETTINGS_H})
+      #
+      # Check whether ADOL-C is configured with extra trig functions
+      #
+      FILE(STRINGS "${ADOLC_SETTINGS_H}" ADOLC_ATRIG_ERF_STRING
+        REGEX "#define ATRIG_ERF")
+      IF(NOT "${ADOLC_ATRIG_ERF_STRING}" STREQUAL "")
+        SET(DEAL_II_ADOLC_WITH_ATRIG_ERF TRUE)
```
Comment (tamiko) [inline cmake/configure/configure_adolc.cmake]: Can you please move the detection of `DEAL_II_ADOLC_WITH_ATRTG_REF` and `DEAL_II_ADOLC_WITH_ADVANCED_BRANCHING` into `FindADOLC.cmake`? Strip the `DEAL_II_` in the process.

In this file please add
```
SET(DEAL_II_ADOLC_WITH_[...] ${ADOLC_[...]})
SET(DEAL_II_ADOLC_WITH_[...] ${ADOLC_[...]})
```

File: include/deal.II/base/config.h.in [hunk 2]
```diff
@@ -140,6 +141,10 @@
  * cmake/modules/Find*.cmake
  */
 
+/* cmake/configure/configure_adolc.cmake */
```
Comment (tamiko) [inline include/deal.II/base/config.h.in]: `/* cmake/modules/FindADOLC.cmake */` after above change.

**Comment by tamiko** [comment]
@jppelteret I sneakingly cleaned up the PR a bit :-] (do not ever `FATAL_ERROR` in a find module 😸)

**Comment by jppelteret** [comment]
@tamiko Thanks a lot! I now see how you intend for the FindPackage module and configuration modules to be used. Now I know for next time ;-)

**Comment by jppelteret** [comment]
In reference to #4704

**Comment by jppelteret** [comment]
Thanks @tamiko , that check LGTM!

File: tests/adolc/basic_functions_jacobian_taped-cpp_vector.cc [hunk 3]
```diff
@@ -0,0 +1,119 @@
...
+#include <adolc/drivers/drivers.h>
+#include <adolc/taping.h>
+
+#include <math.h>
+
+void test_reset_vector_values (const bool reset_values, const int tape_index)
+{
+  const unsigned int m = 5;  // Dependents
+  const unsigned int n = 10; // Independents
+  // std::size_t tape_stats[STAT_SIZE];
```
Comment (tjhei) [inline tests/adolc/basic_functions_jacobian_taped-cpp_vector.cc]: remove

**Comment by jppelteret** [comment]
Fixed and cleaned up the tests, and the Adol-C detection now works as advertised ;-)

**Pull request merged**